tests/expectations/metrics/test_core.py

```diff
@@ -563,6 +563,25 @@ def test_map_unique_pd_column_exists():
     assert list(metrics[condition_metric.id][0]) == [False, False, True, True, False]
     assert metrics[unexpected_count_metric.id] == 2
 
+    unexpected_rows_metric = MetricConfiguration(
+        metric_name="column_values.unique.unexpected_rows",
+        metric_domain_kwargs={"column": "a"},
+        metric_value_kwargs={
+            "result_format": {"result_format": "SUMMARY", "partial_unexpected_count": 1}
+        },
+        metric_dependencies={
+            "unexpected_condition": condition_metric,
+            "table.columns": table_columns_metric,
+        },
+    )
+    results = engine.resolve_metrics(
+        metrics_to_resolve=(unexpected_rows_metric,), metrics=metrics
+    )
+    metrics.update(results)
+
+    assert metrics[unexpected_rows_metric.id]["a"].index == [2]
+    assert metrics[unexpected_rows_metric.id]["a"].values == [3]
+
```

Comment:
What is this test doing? Does it add coverage over this `pandas` issue that was not covered before? Should it be a separate test entirely or are we okay with appending it here?

I don't have the requisite knowledge to definitively say but I thought it was prudent to bring up such questions so you could evaluate.

Reply:
This tests goes into a metric and the result format that triggers the behavior, where the number of rows returned from the pandas dataframe has to be less than its total number of rows.